src/server/gc_worker/gc_worker.rs

```diff
@@ -747,6 +752,9 @@ impl<E: Engine> GcRunnerCore<E> {
             let start_data_key = keys::data_key(start_key.as_encoded());
             let end_data_key = keys::data_end_key(end_key.as_encoded());
 
+            self.coprocessor_hook
+                .pre_delete_range(&start_data_key, &end_data_key);
+
```

Comment:
```suggestion
            self.coprocessor_hook
                .pre_delete_range(start_key.as_encoded(), end_key.as_encoded());
            let start_data_key = keys::data_key(start_key.as_encoded());
            let end_data_key = keys::data_end_key(end_key.as_encoded());

```

Reply:
delete_range in write batch also shares the DeleteRange event in IME, so I think we should keep consistency between them. And `delete_range` in write_batch uses data_key